File: 03/piggy/find_minimum_in_rotated_sorted_array.c
#include <stdio.h>

int findMin(int *, int);

int main(void) {
    int a[7] = {3, 4, 5, 6, 7, 8, 0};
    int min = findMin(a, 7);

    printf("%d\n", min);

    return 0;
}

int findMin(int* nums, int numsSize) {
    int first = 0, last = numsSize - 1, middle = (numsSize - 1) / 2;

    while (first < last) {
        if (nums[first] < nums[last])
            return nums[first];
        
        if (nums[middle] > nums[last]) {
            first = middle + 1;
            middle = (last - first) / 2 + first;
        } else if (nums[middle] < nums[first]) {
            last = middle;
            middle = (last - first) / 2 + first;
        } else {
            break;
        }
    }

    return nums[middle];
}

File: 03/pierpont/Find Minimum in Rotated Sorted Array.java

public class number_ten {
	public static void mian(String[] args) {
		int[] s = {4,5,6,7,0,1,2,3};
		//int m = findMin(s);
		System.out.print("a");
	}
	
    public static int findMin(int[] nums) {
    	int a = nums[0];
    	for(int i = 1;i < nums.length;i++) {
    		if (a > nums[i]) {
    			a = nums[i];
    		}
    	}
		return a;
    }

}
